qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -52,43 +75,145 @@ def _default_options(self):
+            num_qubits(int): The number of qubit in hte circuit.
 
-        return memory
+        Returns:
+            List[List[Tuple[float, float]]]: return a list that each entry is a list that represent a shot.
+            The output structure is  - List[shot index][qubit index] = [I,Q]
+        """
+        # Randomize samples
+        qubits_iq_rand = []
+        for _ in range(shots):
+            rand_i = np.squeeze(np.array(self._get_normal_samples_for_shot(num_qubits)))
+            rand_q = np.squeeze(np.array(self._get_normal_samples_for_shot(num_qubits)))
+            qubits_iq_rand.append(np.array([rand_i, rand_q], dtype="float").T)
 
-    @abstractmethod
-    def _compute_probability(self, circuit: QuantumCircuit) -> float:
-        """Compute the probability used in the binomial distribution creating the IQ shot.
+        memory = []
+        shot_num = 0
+        iq_centers = self._iq_cluster_centers
+
+        for output_number, number_of_occurrences in enumerate(self._rng.multinomial(shots, prob, size=1)[0]):
+            # For multiple qubits - translate a number to a string
+            # and then count them.
+            # need to think about the structure of probability.
+            state_str = str(format(output_number, 'b').zfill(num_qubits))
+            for _ in range(number_of_occurrences):
+                shot_memory = []
+                # the iteration on the string variable state_str starts from the MSB. For readability, we will reverse
+                # the string so the loop will run from the LSB to MSB.
+                for iq_center, qubit_iq_rand_sample, char_qubit in zip(iq_centers, qubits_iq_rand[shot_num], state_str[::-1]):
+                    # The structure of iq_centers is [qubit_number][logic_result][I/Q].
+                    i_center = iq_center[int(char_qubit)][0]
+                    q_center = iq_center[int(char_qubit)][1]
+                    point_i = i_center + qubit_iq_rand_sample
+                    point_q = q_center + qubit_iq_rand_sample
+                    
+                    # Adding phase if not 0.0
+                    if not np.allclose(phase, 0.0):
+                        complex_iq = (point_i + 1.0j * point_q) * np.exp(1.0j * phase)
+                        point_i, point_q = np.real(complex_iq), np.imag(complex_iq)
+                    
+                    shot_memory.append([point_i, point_q])
+                # We proceed to the next occurrence - meaning it's a new shot.
+                memory.append(shot_memory)
+                shot_num += 1
 
-        An abstract method that subclasses will implement to create a probability of
-        being in the excited state based on the received quantum circuit.
+        return memory
 
+    def _generate_data(self, prob_dict: Dict[str, float], num_qubits: int, circuit: QuantumCircuit) -> Dict[str, Union[Union[dict[str, Any], list[list[list[Union[float, complex]]]]], Any]]:
```

Comment:
Yea. I changed it to the following:
```python
    def _generate_data(self, prob_dict: Dict[str, float], num_qubits: int, circuit: QuantumCircuit) \
            -> Dict[str, Union[Dict[str, int], List[List[List[Union[float, complex]]]]]]:
```

the reason for the verity of formats of the output is because the data is organized differently if the user is wants level 1 measurement or  level 2 measurement. I formatted the output to be the same as the output from executing the code on a physical quantum computer so I don't know what can I do to change the signature\output.
Do you have any suggestions?

Reply:
Then I think it's better to just set the type hints to `Any`.

Reply:
OK 😄. I changed it to:
```python
def _generate_data(self, prob_dict: Dict[str, float], num_qubits: int, circuit: QuantumCircuit) -> Dict[str, Any]:
```